core/src/main/java/org/elasticsearch/search/aggregations/bucket/filters/FiltersAggregatorBuilder.java

```diff
@@ -81,6 +92,50 @@ public FiltersAggregatorBuilder(String name, QueryBuilder<?>... filters) {
     }
 
     /**
+     * Read from a stream.
+     */
+    public FiltersAggregatorBuilder(StreamInput in) throws IOException {
+        super(in, InternalFilters.TYPE);
+        keyed = in.readBoolean();
+        int filtersSize = in.readVInt();
+        filters = new ArrayList<>(filtersSize);
+        if (keyed) {
+            for (int i = 0; i < filtersSize; i++) {
+                filters.add(KeyedFilter.PROTOTYPE.readFrom(in));
+            }
+        } else {
+            for (int i = 0; i < filtersSize; i++) {
+                filters.add(new KeyedFilter(String.valueOf(i), in.readQuery()));
+            }
+        }
+        otherBucket = in.readBoolean();
+        otherBucketKey = in.readString();
+    }
+
+    @Override
+    protected void doWriteTo(StreamOutput out) throws IOException {
+        out.writeBoolean(keyed);
+        if (keyed) {
+            out.writeVInt(filters.size());
+            for (KeyedFilter keyedFilter : filters) {
+                keyedFilter.writeTo(out);
+            }
+        } else {
+            out.writeVInt(filters.size());
```

Comment:
very minor and I know this wasn't your code anyway but we could move writing the size out of the if block since it's the same on both branches